RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorOnGPU.cc

```diff
@@ -274,37 +280,35 @@ void CAHitNtupletGeneratorOnGPU<TrackerTraits>::endJob() {
 }
 
 template <typename TrackerTraits>
-PixelTrackHeterogeneousT<TrackerTraits> CAHitNtupletGeneratorOnGPU<TrackerTraits>::makeTuplesAsync(
+TrackSoAHeterogeneousDevice<TrackerTraits> CAHitNtupletGeneratorOnGPU<TrackerTraits>::makeTuplesAsync(
     HitsOnGPU const& hits_d, float bfield, cudaStream_t stream) const {
   using HelixFitOnGPU = HelixFitOnGPU<TrackerTraits>;
-  using PixelTrackHeterogeneous = PixelTrackHeterogeneousT<TrackerTraits>;
+  using TrackSoA = TrackSoAHeterogeneousDevice<TrackerTraits>;
   using GPUKernels = CAHitNtupletGeneratorKernelsGPU<TrackerTraits>;
 
-  PixelTrackHeterogeneous tracks(cms::cuda::make_device_unique<OutputSoA>(stream));
-
-  auto* soa = tracks.get();
-  assert(soa);
+  TrackSoA tracks(stream);
   cudaCheck(cudaGetLastError());
```

Comment:
I cannot find this any more in the latest version of the code, so I'm not sure what the comment was actually referring to...

In general, I think that calling
```c++
cudaCheck(cudaGetLastError())
```
after each kernel launch is good practice, as it can help catch errors in the kernel launch itself (not in the kernel _execution_ though), for example due to invalid parameters or grid size.

In the long term (hopefully not too long) it doesn't matter, since Alpaka does this check internally.

Reply:
I fully agree calling `cudaCheck(cudaGetLastError())` after each kernel is a good practice. I commented only places where it was called after memory allocation, copy, or other CUDA API function that returns an error code (that is checked with `cudaCheck()`).